### PR TITLE
feat(atomic): added placeholder for table result list

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-912

Can be tested [here](http://kit-912.herokuapp.com/) by changing the values of the comboboxes at the top left corner of the screen.

image

### CODE REVIEW

File: packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    --padding: 2rem;
+  }
+
+  &.density-normal {
+    --padding: 1.5rem;
+  }
+
+  &.density-compact {
+    --padding: 1rem;
+  }
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss:53]: Until now, densities were changing the padding directly, but now I introduced a variable.

File: packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    padding: 1rem var(--padding);
+
+    &:first-child {
+      border-top-left-radius: var(--atomic-border-radius-xl);
+    }
+
+    &:last-child {
+      border-top-right-radius: var(--atomic-border-radius-xl);
+    }
+  }
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss:34]: These border radius weren't using a variable until now.

File: packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  margin: 2rem 0;
+  border-spacing: 0;
+  border-collapse: separate;
+
+  th,
+  td {
+    font-family: var(--atomic-font-family);
+    border: 1px solid var(--atomic-neutral);
+    --font-size: var(--atomic-text-sm);
+    font-size: var(--font-size);
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss:12]: This font size wasn't using a variable until now.

File: packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss [hunk 4]
```diff
@@ -0,0 +1,54 @@
+@define-mixin atomic-result-table {
+  min-width: 100%;
+  margin: 2rem 0;
+  border-spacing: 0;
+  border-collapse: separate;
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-result-list-v1/atomic-result-table.pcss:5]: `border-collapse` wasn't needed until now. Its default value is `separate` in the usual context, but in a shadowDOM with Tailwind it seems to be `collapse`.

File: packages/atomic/src/components/atomic-result-placeholder-v1/atomic-result-placeholder.tsx [hunk 5]
```diff
@@ -10,27 +10,17 @@ const placeholderClasses = 'block bg-neutral w-full h-full rounded';
...
-   */
-  @Prop() density: ResultDisplayDensity = 'normal';
-
-  /**
-   * How large or small the visual section of results should be.
-   */
-  @Prop() image: ResultDisplayImageSize = 'icon';
+  @Prop() display!: ResultDisplayLayout;
+  @Prop() density!: ResultDisplayDensity;
+  @Prop() image!: ResultDisplayImageSize;
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-result-placeholder-v1/atomic-result-placeholder.tsx:23]: These comments weren't necessary so I removed them.

Comment (olamothe) [reply packages/atomic/src/components/atomic-result-placeholder-v1/atomic-result-placeholder.tsx:23]: Why do you say they weren't necessary ?

**Review by olamothe — APPROVED** [review]
Great looking !

However, when I look at the final product here ( https://z4mou.csb.app/ , which should contain all the latest changes you did for placeholders in the last few days ?), there's something weird looking when it's doing "the full loop" with placeholders.

Place holder appears on load, then disappears when the query is finished (so we get a "blank" page for a fraction of a second), then results displays.

The ideal scenario would be that we remove the placeholders only when results are ready to display/loaded/"rendered" (but hidden). So that we do not get that weird blank page state.

**Comment by btaillon** [comment]
> Place holder appears on load, then disappears when the query is finished (so we get a "blank" page for a fraction of a second), then results displays.
>
> — @olamothe 

This seems to be a script performance issue with a bit of rendering lag that lasts about 560ms, I'll investigate it in a separate story.
image

File: packages/atomic/src/components/atomic-result-placeholder-v1/atomic-result-placeholder.tsx (hunk 5 above)
Comment (btaillon) [reply packages/atomic/src/components/atomic-result-placeholder-v1/atomic-result-placeholder.tsx:23]: Because the placeholder component is internal.

**Review by samisayegh — APPROVED** [review]
Really nice demo w/ toggles 💯

**Pull request merged**